modules/rest_api/controllers/services/rest.php

```diff
@@ -1130,6 +1130,41 @@ private function getColumnsTemplate(&$postObj) {
       $this->esCsvTemplateRemoveColumns = (array) $postObj->removeColumns;
       unset($postObj->removeColumns);
     }
+    if (isset($postObj->columnsSurveyId)) {
+      $this->esCsvSurveySampleColumns = $this->getSurveyAttributes('sample', $postObj->columnsSurveyId);
+      $this->esCsvSurveyOccurrenceColumns = $this->getSurveyAttributes('occurrence', $postObj->columnsSurveyId);
+      unset($postObj->columnsSurveyId);
+    }
+  }
+
+  /**
+   * Retrieve the columns associated with a survey.
+   *
+   * @param string $type
+   *   Either 'sample' or 'occurrence'.
+   * 
+   * @param string $id
+   *   The survey ID.
+   * 
+   * @return array
+   *   Array of associative arrays describing each attribute.
+   */
+  private function getSurveyAttributes($type, $id) {
```

Comment:
As this will be (nearly) static caching might be worth considering.

Reply:
@johnvanbreda I don't know how to implement static caching - is there an example I can look at?

Reply:
There's an example in this commit - it's built into Kohana: https://github.com/Indicia-Team/warehouse/commit/b70096345ba3a733328db8cba929927b486ef81c

Reply:
I've had a go at this @johnvanbreda and it seems to work. I'm not tagging the cache. Is tagging used so that a chache can be explicitly cleared elsewhere. Shoul I be tagging it?